### PR TITLE
Update Custom Modules Latest to include header and class name changes.

### LINKED ISSUE #5204: Outdated header files and class names on custom modules in c++ page
Latest version https://docs.godotengine.org/en/latest/development/cpp/custom_modules_in_cpp.html

Change:
#include "core/class_db.h" --> #include "core/object/class_db.h
#include "core/reference.h" --> #include "core/object/ref_counted.h" 
`class Summator : public Reference {
    GDCLASS(Summator, Reference);` -->
`class Summator : public RefCounted {
    GDCLASS(Summator, RefCounted);`

I'm not sure if this entirely fixes all of the problems or is correct because I still can't quite get the changes working for me otherwise I'd make a pull request.

### PR DESCRIPTION
**Pull request by allensam**
Closes #5204 and https://github.com/godotengine/godot/issues/50741

### CODE REVIEW

**Comment by allensam** [comment]
Actually I should go back and make changes for the rest of the c++ modules docs

**Comment by pycbouh** [comment]
> Actually I should go back and make changes for the rest of the c++ modules docs

That's a good idea. I'll convert the PR to a draft in the meantime.

**Comment by mhilbrunner** [comment]
@allensam Thanks for contributing! Can you fix the things noted above? :)

File: development/cpp/custom_modules_in_cpp.rst [hunk 1]
```diff
@@ -129,7 +129,7 @@ These files should contain the following:
 
     #include "register_types.h"
 
-    #include "core/class_db.h"
+    #include "core/object/class_db.h
```
[the remark this answers is not in the document]
Comment (akien-mga) [reply development/cpp/custom_modules_in_cpp.rst:132]:
```suggestion
    #include "core/object/class_db.h"
```

File: development/cpp/custom_modules_in_cpp.rst [hunk 2]
```diff
@@ -366,7 +366,7 @@ We now need to add this method to ``register_types`` header and source files:
...
```
Comment (akien-mga) [inline development/cpp/custom_modules_in_cpp.rst:369]:
```suggestion
    #include "core/object/class_db.h"
```

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**